src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/base.py

```diff
@@ -159,7 +162,7 @@ def _process_request_recording(self, request):
             return None
 
         if self.in_recording:
-            for processor in self.recording_processors:
+            for processor in self.recording_processors + self.recording_post_processors:
```

Comment:
We can't add `ContentLengthProcessor` here

https://github.com/Azure/azure-cli/blob/30216d0268f3491b8af2aa060d90a9e1f94f9b3c/src/azure-cli-testsdk/azure/cli/testsdk/base.py#L89-L97

because `AbstractPreparer` will add preparers when **executing the test method**

https://github.com/Azure/azure-cli/blob/30216d0268f3491b8af2aa060d90a9e1f94f9b3c/src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/preparers.py#L34

causing these newly added preparers to be called **after** `ContentLengthProcessor`